Source/Core/Core/State.cpp

```diff
@@ -378,48 +395,12 @@ static void CompressAndDumpState(CompressAndDumpState_args& save_args)
     return;
   }
 
-  // Setting up the header
-  StateHeader header{};
-  SConfig::GetInstance().GetGameID().copy(header.gameID, std::size(header.gameID));
-  header.size = s_use_compression ? (u32)buffer_size : 0;
-  header.time = GetSystemTimeAsDouble();
-
-  f.WriteArray(&header, 1);
-
-  if (header.size != 0)  // non-zero header size means the state is compressed
-  {
-    lzo_uint i = 0;
-    while (true)
-    {
-      lzo_uint32 cur_len = 0;
-      lzo_uint out_len = 0;
+  WriteHeadersToFile(buffer_size, f);
 
-      if ((i + IN_LEN) >= buffer_size)
-      {
-        cur_len = (lzo_uint32)(buffer_size - i);
-      }
-      else
-      {
-        cur_len = IN_LEN;
-      }
-
-      if (lzo1x_1_compress(buffer_data + i, cur_len, out, &out_len, wrkmem) != LZO_E_OK)
-        PanicAlertFmtT("Internal LZO Error - compression failed");
-
-      // The size of the data to write is 'out_len'
-      f.WriteArray((lzo_uint32*)&out_len, 1);
-      f.WriteBytes(out, out_len);
-
-      if (cur_len != IN_LEN)
-        break;
-
-      i += cur_len;
-    }
-  }
-  else  // uncompressed
-  {
+  if (s_use_compression)
+    CompressBufferToFile(buffer_data, buffer_size, f);
+  else
```

Comment:
Sorta pre-existing but since you're here... if either the compression or the write fails, this should probably do something about that instead of blindly continuing the savestate write. We can do that in another PR since it's not new, if you want.

Reply:
I'll do this in a new PR in the near future if that's alright!